### ISSUE TITLE
Player fails on mobile (Android)

### ISSUE THREAD

**Issue by klausw** [shimberger/gohls#73]
I've set this up, and it works great when using Chrome on Windows, but I'm having trouble getting it working on an Android phone. I'm getting this error in the video player:

_The media could not be loaded, either because the server or network failed or because the format is not supported._

Using USB remote inspector, I see the following on the JS console:

```
Uncaught (in promise) DOMException: play() failed because the user didn't interact with the document first. https://goo.gl/xX8pDD
2.70700d14.chunk.js:formatted:16904 VIDEOJS: ERROR: (CODE:4 MEDIA_ERR_SRC_NOT_SUPPORTED) The media could not be loaded, either because the server or network failed or because the format is not supported. Zn {code: 4, message: "The media could not be loaded, either because the …rk failed or because the format is not supported."}
```

I'm not sure if the play() DOMException is the root cause - this appears to be due to autoplay being forbidden in recent Chrome releases?

Here's the DEBUG=1 log when I run this:

```
2021/04/30 19:16:08 "GET http://192.168.114.74:8080/api/item/ HTTP/1.1" from 192.168.114.135:49654 - 200 772B in 56.603µs
DEBU[0009] Executing ffprobe for /tmp/vidtest/test2.mkv 
DEBU[0009] ffprobe for /tmp/vidtest/test2.mkv returned  
2021/04/30 19:16:13 "GET http://192.168.114.74:8080/api/item/8dbfd711761b3a8af64ccbd5eafb86eda1fed261 HTTP/1.1" from 192.168.114.135:49654 - 200 501B in 44.063631ms
DEBU[0009] Executing: ffmpeg [-timelimit 15 -loglevel error -ss 5.0 -i /tmp/vidtest/test2.mkv -vf scale=320:-1 -frames:v 1 -f image2 -] 
2021/04/30 19:16:13 "GET http://192.168.114.74:8080/api/frame/00f468a890aa6cc897c16e42435a70823b39808f?t=5 HTTP/1.1" from 192.168.114.135:49654 - 200 8575B in 75.740204ms
2021/04/30 19:16:19 "GET http://192.168.114.74:8080/api/item/00f468a890aa6cc897c16e42435a70823b39808f HTTP/1.1" from 192.168.114.135:49654 - 200 435B in 64.093µs
2021/04/30 19:16:19 "GET http://192.168.114.74:8080/api/playlist/00f468a890aa6cc897c16e42435a70823b39808f HTTP/1.1" from 192.168.114.135:49654 - 200 1196B in 224.436µs
2021/04/30 19:16:19 "GET http://192.168.114.74:8080/api/playlist/00f468a890aa6cc897c16e42435a70823b39808f HTTP/1.1" from 192.168.114.135:49656 - 200 1196B in 173.185µs
2021/04/30 19:16:19 "GET http://192.168.114.74:8080/api/playlist/00f468a890aa6cc897c16e42435a70823b39808f HTTP/1.1" from 192.168.114.135:49658 - 200 1196B in 235.498µs
```

When I open the same file from the desktop browser, I get transcoding debug entries such as the one below, I don't get those when I try from the Android browser. It looks as if it doesn't get to the point of actually trying to play the video in the Android browser?

```
DEBU[0061] Executing: ffmpeg [-timelimit 45 -ss 0.00 -i /tmp/vidtest/test2.mkv -t 5.00 -vf scale=-2:720 -vcodec libx264 -preset veryfast -c:a aac -b:a 128k -ac 2 -pix_fmt yuv420p -force_key_frames expr:gte(t,n_forced*5.000) -f ssegment -segment_time 5.00 -initial_offset 0.00 pipe:out%03d.ts] 
```

The video file for this test is https://github.com/Matroska-Org/matroska-test-files/blob/master/test_files/test2.mkv?raw=true , and I'm running it as follows with a fresh ffmpeg static build from https://johnvansickle.com/ffmpeg/ :

```
PATH=ffmpeg-git-20210425-amd64-static:$PATH DEBUG=1 ./gohls --listen :8080 serve
```

Is there anything else I could do to help debug this? It seems like a great solution for accessing my video library using an Android tablet on my exercise bike, without the hassle of setting up a heavyweight streaming server, so I hope this is fixable :-)

**Comment by shimberger** [comment]
Hi @klausw sorry for the late reply. Are you still interested in a fix? I was very busy at work but have some time this week.

**Comment by klausw** [comment]
Yes, I am still interested. Please let me know if there are things to try
to help debug this.

**Comment by shimberger** [comment]
@klausw Great. Could you get me some info on your Android Version and also the browser version? 

https://www.wikihow.com/Find-Your-Version-on-Chrome-on-Android
https://its.uiowa.edu/support/article/2803

It might be an issue with the browser not supporting HLS or some video format.

I also created a new snapshot with updated JS libraries yesterday. You can find the URL in the readme. maybe that helps already.

**Comment by klausw** [comment]
I tried today's snapshot, and it's working now. Thank you!!!

This is using Chrome 91.0.4472.134 and Chrome Canary 92.0.4504.2. OS is Android 11/RQ3A on a Pixel 5.

Before upgrading, I confirmed that the earlier version I had downloaded (a snapshot dated Nov 5 2020 based on the file's timestamp) didn't work on both of those Chrome version.

(Apologies for not including version numbers earlier - I grumble at users who don't do that myself ;-)  Maybe consider adding a "new issue" template to this repo which asks for that information?)
